sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -207,9 +207,11 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
         className = table.provider.get,
         options = table.storage.properties)
 
-    LogicalRelation(
+    val logicalRel = LogicalRelation(
       dataSource.resolveRelation(),
       metastoreTableIdentifier = Some(table.identifier))
+    logicalRel.inheritedStats = table.catalogStats
```

Comment:
Do you mean adding statistics as a constructor parameter in LogicalRelation? I tried this in my first implementation, but many code paths needed to be fixed (pattern match like LogicalRelation(src: BaseRelation, _, _)). I'll switch back if that's ok.